File: Solutions/Solution121.java
class Solution121 {
  public int maxProfit(int[] prices) {
    // For an integer array, find the maximum range between n and m where n[ind] < m[ind]
    int low = Integer.MAX_VALUE;
    int highProfit = 0;

    for (int i = 0; i < prices.length; i++) {
      if (prices[i] < low) {
        low = prices[i];
      } else if (prices[i] - low > highProfit) {
        highProfit = prices[i] - low;
      }
    }
    return highProfit;
  }
}

File: Solutions/Solution121.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        maxProfit = 0

        low = prices[0]
        for i in range(len(prices)):
            if prices[i] < low:
                low = prices[i]

            if prices[i] - low > maxProfit:
                maxProfit = prices[i] - low

        return maxProfit

        
        